pkg/v1/tkg/tkgconfigbom/bom.go

```diff
@@ -337,14 +337,29 @@ func (c *client) IsCustomRepositorySkipTLSVerify() bool {
 	return false
 }
 
-func (c *client) GetCustomRepositoryCaCertificate() ([]byte, error) {
-	value, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableCustomImageRepositoryCaCertificate)
-	if err != nil {
-		// return empty content when not specified
+// GetCustomRepositoryCaCertificateForClient returns CA certificate to use with cli client
+// This function reads the CA certificate from following variables in below orders:
+// 1. PROXY_CA_CERT
+// 2. TKG_PROXY_CA_CERT
+// 3. TKG_CUSTOM_IMAGE_REPOSITORY_CA_CERTIFICATE
+func (c *client) GetCustomRepositoryCaCertificateForClient() ([]byte, error) {
+	caCert := ""
+	value1, err1 := c.TKGConfigReaderWriter().Get(constants.ProxyCACert)
```

Comment:
nit: more readable if we name the valueX variables appropriately. proxyCACertValue, tkgProxyCACertValue, etc.

I suggest changing the comment to say that the variables are used in decreasing order of precedence or something like that.

Reply:
Updated